src/main/java/org/snomed/snowstorm/rest/ExportController.java

```diff
@@ -59,10 +73,33 @@ public ExportConfiguration getExportJob(@PathVariable String exportId) {
 	@GetMapping(value = "/{exportId}/archive", produces="application/zip")
 	public void downloadRf2Archive(@PathVariable String exportId, HttpServletResponse response) throws IOException {
 		ExportConfiguration exportConfiguration = exportService.getExportJobOrThrow(exportId);
+		if (!exportConfiguration.isStartExport()) {
+			String filename = exportService.getFilename(exportConfiguration);
+			response.setHeader("Content-Disposition", "attachment; filename=\"" + filename + "\"");
+			exportService.exportRF2Archive(exportConfiguration, response.getOutputStream());
+		} else {
+			// First download will have deleted the file
+			if (Objects.equals(ExportStatus.DISCARDED, exportConfiguration.getStatus())) {
+				response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+				response.getWriter().write("Archive has already been downloaded; cannot re-download");
+				response.getWriter().flush();
+				return;
+			}
+
+			// Initial check before polling
+			if (Objects.equals(ExportStatus.PENDING, exportConfiguration.getStatus()) || Objects.equals(ExportStatus.IN_PROGRESS, exportConfiguration.getStatus())) {
+				exportConfiguration = pollExportStatus(exportConfiguration);
```

Comment:
@dmcgihtsdo If you insist to use the workaround (i.e with startExport flag) then we don't need to add backward compatibility here. You should respond with bad request when export is not completed.

Reply:
Sorry but I'm not sure what you mean @CoderMChu. 

`exportRF2ArchiveAsync & copyRF2Archive` is backwards compatible, but for testing capacity, I've kept the use of `exportRF2Archive`.

Reply:
@dmcgihtsdo We don't need to pollExportStatus() when you have used startExport flag for backward compatibility.  The correct API call should have checked the status before calling download api. (i.e The client should poll)

Reply:
Thanks for the clarification @CoderMChu. I have made that change.